### DISCUSSION TITLE
Absence time off

### DISCUSSION THREAD

**Discussion by DaveeeKR** [kimai/Paid-plugins#131] [Questions & Answers]
Hello,

I have a question about the Working Contract plugin and Time Off.

As a company, we were unable to update the production machine at the start of the year, and we would like to make an update from the 1st of March. Is there any possible way to start counting time off in absence from this date? To be more precise, we would like to (somehow) edit the algorithm to begin counting "time off" not from the beginning of the year, but from the 1st of March, excluding January and February. If this is possible, how can we revert it for the proper count in 2025?

**Comment by kevinpapst** [comment]
Like that one?

![image](https://github.com/kevinpapst/kimai-plugins/assets/533162/3e100cb8-9302-4943-92f9-8020a0d8eaf8)

It prevents the calculation before the entered date

**Reply by jtechwyr** [reply]
Yes, that's exactly it. Is there a way to set this parameter globally for all users? If not, could you please specify in which database table it is stored?

**Reply by DaveeeKR** [reply]
I have tested it on my local test machine, and its not working correctly. I have filled callendar from the begining of the year until now, set "working since" for today, and it shows me i have balance of 216 hours.

![image](https://github.com/kevinpapst/kimai-plugins/assets/129747290/a604246c-ea76-47ed-9248-35c1f67a73bb)
![image](https://github.com/kevinpapst/kimai-plugins/assets/129747290/5622b001-269e-4ac1-829c-f1432b7a6e85)

**Reply by kevinpapst** [reply]
Your use-case just doesn't make sense to me. 
Do you expect that the logged time gets lost? 
If you have time in there, you don't need that setting. 

@jtechwyr you cannot set it globally, it is per user. there is one row per user in kimai2_user_preferences

**Reply by jtechwyr** [reply]
As DaveeeKR stated, we have users who have entries in January and February. When we set the "Working since" date to March 1st, all entries from January and February are considered overtime for each of these users. Can we change this so that these entries are not counted as "Time off"?

We understand that one option would be to delete the entries for these users from those months, but then we would lose the history.

**Reply by kevinpapst** [reply]
By now timesheets always count as working-time. 
I don't understand why you would set it to March, when you have the times from January.

**Reply by jtechwyr** [reply]
Currently, we have a lower version of Kimai, in which we use a different plugin to calculate overtime. Upon updating, these overtime hours do not transfer because the new version uses a different plugin for this purpose. By continuing to use the old version until the end of February, we will reset overtime hours for users, for example, by compensating them. Therefore, we do not want the update to count entries from January and February as overtime, as users have already been compensated for them. We also do not want to delete entries because we need to have access to them.

**Reply by kevinpapst** [reply]
Understand now, thanks for the explanation!
In that case, it is best to set the date as discussed to first of March and then use the "manual booking" to create compensation entries to account for that "overtime" that Kimai shows.

For example "User A" has 94 hours for January and February. Then you create a "manual entry" with `-94 hours`:
<img width="786" alt="Bildschirmfoto 2024-02-12 um 10 22 12" src="https://github.com/kevinpapst/kimai-plugins/assets/533162/4a65bbd1-e0bc-46a7-aed9-8800d620d7fa">

<img width="537" alt="Bildschirmfoto 2024-02-12 um 10 22 54" src="https://github.com/kevinpapst/kimai-plugins/assets/533162/4520b05c-b339-4fd4-9058-b9bafb807b17">

**Reply by jtechwyr** [reply]
Sure, is there a way to bypass this, to automate it somehow? We have around 250 users, and it would take quite a bit of time to do it manually.

**Reply by kevinpapst** [reply]
On which number is the compensation based? Is it somewhere available in the database?

**Reply by jtechwyr** [reply]
Do I understand correctly that you're asking whether in the old version, in the old plugin, there's a number in the database indicating the number of overtime hours for a given user?
